subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -453,6 +454,17 @@ public void setDebug(boolean enabled) {
         forkOptions.setDebug(enabled);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setDebugOptions(JavaDebugOptions debugOptions) {
+        forkOptions.setDebugOptions(debugOptions);
+        // run all tests on the same JVM
```

Comment:
I don't think we need to do this.  

forkEvery is "OK" to leave as-is, it just means we'll start a new JVM on every test. 

maxParallelForks is already set to 1 when debug = true.

Reply:
Well,  I think when the user tries to debug tests then the expectation is to attach a debugger only once. I think setting those properties make sense (we were right though, `maxParallelForks` was already set; I removed that from the code).